### PR TITLE
[tools/tflitefile] Support controlflow operators in select_operator.py

### PR DESCRIPTION
**Pull request by ragmani**
This commit make select_operator.py supports controlflow operators
  - Support child subgraphs
  - Add generating builtin options of controlflow operators

Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: tools/tflitefile_tool/select_operator.py [hunk 1]
```diff
@@ -63,7 +63,44 @@ def GetOperatorList(oplist_file):
     return opcode_list
 
 
-def GenerateOperatorCodes(new_builder, sample_model, used_operators_dic):
+def GetUsedSubgraphsList(sample_model, subg_num, opcode_list, used_subgraphs_list):
```
Comment (glistening) [inline tools/tflitefile_tool/select_operator.py:66]: @ragmani As I understand your PR, `opcode_list` here should be renamed to `operator_list` also.

Comment (ragmani) [reply tools/tflitefile_tool/select_operator.py:66]: Right, I will change the name. Thanks for your pointing out.

File: tools/tflitefile_tool/select_operator.py [hunk 2]
```diff
@@ -1120,30 +1213,49 @@ def main(args):
...
                 # default: same as input sample
                 used_tensors.append(input_tensor_idx)
 
-        for output_idx in range(opcode.OutputsLength()):
-            output_tensor_idx = opcode.Outputs(output_idx)
+        for output_idx in range(operator.OutputsLength()):
+            output_tensor_idx = operator.Outputs(output_idx)
             if not output_tensor_idx in used_tensors:
                 # default: same as input sample
                 used_tensors.append(output_tensor_idx)
```
Comment (glistening) [inline tools/tflitefile_tool/select_operator.py:1241]: @ragmani I think `used_tensors` could be updated with the tensors used in subgraphs from `If` and `While`.

Comment (ragmani) [reply tools/tflitefile_tool/select_operator.py:1241]: `used_tensors` just has tensors a selected subgraph.

Comment (glistening) [reply tools/tflitefile_tool/select_operator.py:1241]: @ragmani You mean that it is guaranteed that no additional `used_tensors` will be used except for those by _selected operators of a selected subgraph_? If so, it is okay.

Comment (ragmani) [reply tools/tflitefile_tool/select_operator.py:1241]: (ADDED) Those updated buffers of subgraphs to `used_buffers` are not calculated from `used_tensors`. The buffers are not related to tensors of `used_tensors`.

**Comment by ragmani** [comment]
@nnfw-bot test cross-aarch64-runtime-test-release

**Comment by ragmani** [comment]
@nnfw-bot test cross-runtime-test-release

File: tools/tflitefile_tool/select_operator.py (hunk 2 above)
Comment (ragmani) [reply tools/tflitefile_tool/select_operator.py:1241]: @glistening 
The code is to add buffers of tensors of selected subgraphs. But missing buffers were in other child subgraphs branched by control flow operations. So it was necessary to add the buffers of the child subgraphs to `used_buffers`. And I did it like below
```python
    # Append buffers of tensors of child subgraphs
    for subgraph_idx in used_subgraphs_list:
        if subgraph_idx == subgraph:
            continue
        for tensor_idx in range(sample_model.Subgraphs(subgraph_idx).TensorsLength()):
            tensor = sample_model.Subgraphs(subgraph_idx).Tensors(tensor_idx)
            used_buffers.append(tensor.Buffer())
```
The buffers are not related to tensors of `used_tensors`.

Comment (glistening) [reply tools/tflitefile_tool/select_operator.py:1241]: @ragmani 

> The code is to add buffers of tensors of selected subgraph. But missing buffers were in other child subgraphs branched by control flow operations

Definitely I know. It is why I wrote 

> If you are going to update used_buffer, it is okay.

**Comment by ragmani** [comment]
PTAL @glistening @wateret @hyunsik-yoon

**Comment by hyunsik-yoon** [comment]
It would be better if we refine `README.md` too.

File: tools/tflitefile_tool/select_operator.py [hunk 3]
```diff
@@ -992,19 +1060,43 @@ def GenerateSubgraph(new_builder, selected_subgraph, opcode_list, new_input_tens
...
 
-def GenerateSubgraphs(args, new_builder, sample_model, opcode_list, new_input_tensor,
+def GenerateSubgraphs(args, new_builder, sample_model, operator_list, new_input_tensor,
                       new_output_tensor, used_tensors_dic, used_buffers_dic,
-                      used_operators_dic):
+                      used_opcodes_dic, used_subgraphs_dic):
+
     new_subgraph_list = []
 
     # We think only one subgraph
```
Comment (hyunsik-yoon) [inline tools/tflitefile_tool/select_operator.py:1069]: maybe not relevant anymore?

Comment (ragmani) [reply tools/tflitefile_tool/select_operator.py:1069]: Thanks. I will remove it.

**Comment by ragmani** [comment]
It would be better if we refine README.md too.

@hyunsik-yoon I refined it.

**Pull request merged**